### PR TITLE
Add documentation for zlib

### PR DESCRIPTION
**Pull request by janaknat**
*Issue #, if available:*

*Description of changes:*
Describe zlib, its many forks and the work done to improve decompression performance in zlib-cloudflare for ARM and x86 platforms.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: zlib.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Performance of ZLIB forks for inflate and deflate
+
+ZLIB is a compression library used to compress and decompress data by popular software such as Java packages, package management using zip, gzip, png etc. Due to its wide usage only rare changes are accepted, which has lead to numerous forks - same interface with improved functionality. Of the many, there are 4 predominantly used zlib forks:
```
Comment (AGSaidi) [inline zlib.md:3]: overall this is too wordy. All we want to say is that for the best perforance on Graviton2 please use the zlib-cloudflare fork and just put it in the generic table we have of libarries.

Comment (janaknat) [reply zlib.md]: Ok. I'll update the table with an entry and remove zlib.md
